File: Practica_1/src/gestorAplicacion/noPersonas/OrdenVirtual.java
package gestorAplicacion.noPersonas;
import java.util.ArrayList;

import gestorAplicacion.personas.*;
import uiMain.ActivarPromocion;


public class OrdenVirtual implements Orden{

	private static ArrayList<OrdenVirtual> ordenesVirtuales = new ArrayList<OrdenVirtual>();
	private static int nroOrdenes = 2000;
	private ArrayList<Bebida> bebidas = new ArrayList<Bebida> ();
	private Cliente cliente;
	private Barista barista;
	private Domiciliario domiciliario;
	private int nroOrden;
	private float costo;
	private static int ordenesCreadas;
	
	public OrdenVirtual(ArrayList<Bebida> bebidas, Cliente cliente, Barista barista, Domiciliario domiciliario) {
		this();
		this.nroOrden = nroOrdenes;
		nroOrdenes ++;
		this.bebidas = bebidas;
		this.cliente = cliente;
		this.barista = barista;
		this.domiciliario = domiciliario;
		float acumulado = 0;
		for (Bebida x: bebidas) acumulado += x.getPrecio();
		this.costo = acumulado;
		ordenesVirtuales.add(this);
	}
	
	public static int getOrdenesCreadas() {
		return ordenesCreadas;
	}

	public static void setOrdenesCreadas(int ordenesCreadas) {
		OrdenVirtual.ordenesCreadas = ordenesCreadas;
	}

	public OrdenVirtual() {
		ordenesCreadas ++;
	}

	public void aplicarPromocion(Promocion promocion) {
		if(promocion.equals(Promocion.CAFE2X1) && ActivarPromocion.isPromos()) {
			this.costo -= 5000;
		} else if(promocion.equals(Promocion.JUGO2X1) && ActivarPromocion.isPromos()) {
			this.costo -= 4000;
		} else if (promocion.equals(Promocion.CAFECONJUGOAL50) && ActivarPromocion.isPromos()){
			this.costo -= 4500;
		}
	}
	
public void reportarIncidente() {
		
		this.domiciliario.setIncidentes(true);
	}

	public String toString() {
		return this.barista.getNombre() + " le vendió a " + this.cliente.getNombre()+" la orden número " + this.nroOrden + " por un valor de " + this.costo;
	}
	
	public static ArrayList<OrdenVirtual> getOrdenesVirtuales() {
		return ordenesVirtuales;
	}

	public static void setOrdenesVirtuales(ArrayList<OrdenVirtual> ordenesVirtuales) {
		OrdenVirtual.ordenesVirtuales = ordenesVirtuales;
	}

	public static int getNroOrdenes() {
		return nroOrdenes;
	}

	public ArrayList<Bebida> getBebidas() {
		return bebidas;
	}

	public Cliente getCliente() {
		return cliente;
	}

	public Barista getBarista() {
		return barista;
	}

	public Domiciliario getDomiciliario() {
		return domiciliario;
	}

	public int getNroOrden() {
		return nroOrden;
	}

	public float getCosto() {
		return costo;
	}

	

	public static void setNroOrdenes(int nroOrdenes) {
		OrdenVirtual.nroOrdenes = nroOrdenes;
	}

	public void setBebidas(ArrayList<Bebida> bebidas) {
		this.bebidas = bebidas;
	}

	public void setCliente(Cliente cliente) {
		this.cliente = cliente;
	}

	public void setBarista(Barista barista) {
		this.barista = barista;
	}

	public void setDomiciliario(Domiciliario domiciliario) {
		this.domiciliario = domiciliario;
	}

	public void setNroOrden(int nroOrden) {
		this.nroOrden = nroOrden;
	}

	public void setCosto(float costo) {
		this.costo = costo;
	}
	
}

File: Practica_2/noPersonas/ordenVirtual.py
from noPersonas.orden import Orden
from noPersonas.promocion import Promocion

class OrdenVirtual(Orden):
    _ordenesVirtuales = []
    _nroOrdenes = 2000
    _ordenesCreadas = 0

    def __init__(self, bebidas, cliente, barista, domiciliario):
        OrdenVirtual._ordenesCreadas += 1
        self._nroOrden = OrdenVirtual._nroOrdenes
        OrdenVirtual._nroOrdenes += 1
        self._bebidas = bebidas
        self._cliente = cliente
        self._barista = barista
        self._domiciliario = domiciliario
        acumulado = 0
        OrdenVirtual._ordenesVirtuales.append(self)
        
        for x in self._bebidas:
            acumulado += x.getPrecio()
        
        self._costo = acumulado
        OrdenVirtual._ordenesVirtuales.append(self)
    
    @classmethod
    def getOrdenesCreadas(cls):
        return cls._ordenesCreadas
    
    @classmethod
    def setOrdenesCreadas(cls, ordenesCreadas):
        cls._ordenesCreadas = ordenesCreadas
        
    def aplicarPromocion(self, promocion):
        if promocion == Promocion.CAFE2X1: # && ActivarPromocion.isPromos():
            self._costo -= 5000
        elif promocion == Promocion.JUGO2X1: # && ActivarPromocion.isPromos():
            self._costo -= 4000
        elif promocion == Promocion.CAFECONJUGOAL50: # && ActivarPromocion.isPromos()):
            self._costo -= 4500
    
    def reportarIncidente(self):
        self._domiciliario.setIncidentes(True)
    
    def __str__(self):
        return self._barista.getNombre() + ' le vendió a ' + self._cliente.getNombre()+' la orden número ' + self._nroOrden + ' por un valor de ' + self._costo
    
    @classmethod
    def getOrdenesVirtuales(cls):
        return cls._ordenesVirtuales
    
    @classmethod
    def setOrdenesVirtuales(cls, ordenesVirtuales):
        cls._ordenesVirtuales = ordenesVirtuales
    
    @classmethod
    def getNroOrdenes(cls):
        return cls._nroOrdenes
    
    def getBebidas(self):
        return self._bebidas
    
    def getCliente(self):
        return self._cliente;
    
    def getBarista(self):
        return self._barista
    
    def getDomiciliario(self):
        return self._domiciliario

    def getNroOrden(self):
        return self._nroOrden

    def getCosto(self):
        return self._costo

    @classmethod
    def setNroOrdenes(cls,nroOrdenes):
        cls._nroOrdenes = nroOrdenes
    
    def setBebidas(self, bebidas):
        self._bebidas = bebidas

    def setCliente(self, cliente):
        self._cliente = cliente

    def setBarista(self, barista):
        self._barista = barista

    def setDomiciliario(self, domiciliario):
        self._domiciliario = domiciliario

    def setNroOrden(self, nroOrden):
        self._nroOrden = nroOrden

    def setCosto(self, costo):
        self._costo = costo